broadinstitute/pearl
# [JN-525] site content populate

Previously, there was no way to *just* update website content via the API, you had to update the whole portal.  This adds an API endpoint for accessing the siteContentPopulator directly.  It also adds a UX for accessing populate functions (visible only to superusers), so that we don't have to futz with the swagger API.

![image](https://github.com/broadinstitute/juniper/assets/2800795/9520af4d-dccf-40e4-ab76-a336284e5064)


This PR only includes populating the new content--it does not handle attaching it to an environment.  That will be a later PR (which will include general version-selecting of siteContent)

I probably already got carried away too much on this ticket, so I had to stop, but there are definitely some things I hope we get to later, including:
1. Making it possible to overwrite a SiteContent object even while it's attached to a portal (similar to how we can do that for surveys) -- this is useful in development to quickly test changes without nuking your whole portal every time
2. some convenience improvements for the UX -- filenames should auto-suggest based on conventions, portal shortcodes could be prepopulated, etc.
3. a simplified "reset demo" button, available only in the demo environment, for Jeff/Mounica to press that resets everything

TO TEST: 

1. redeploy adminApiApp
2. login to the admin tool
3. choose "Populate" from the superuser functions
4. choose "Site content" from the nav
5. enter "ourhealth"  for portal shortcode
6. select "No" for overwrite
7. enter "portals/ourhealth/siteContent/siteContent.json"
8. presss "Populate"
9. confirm the UX displays a success message
10. in your database, `select id, stable_id, version from site_content` to confirm the new version was created
